File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let tmp = new Map()
    for (let i = 0; i< nums.length;i++) {
        const value = nums[i]
        const paired = target - value
        if (tmp.has(paired)){
            return [tmp.get(paired), i]
        }else{
            tmp.set(nums[i],i)
        }
        //console.log(tmp)
    }
    
    return []
};

File: 1-two-sum/1-two-sum.go
func twoSum(nums []int, target int) []int {
    tmp := make(map[int]int)
    result := []int{}
    for i,v := range nums {
        paired := target - v
        if val, ok := tmp[paired]; ok {
            result = append(result,val)
            result = append(result,i)
            return result
        }else{
            tmp[v] = i
        }
    }
    return result
}